silhouette/app/com/mohiva/play/silhouette/impl/providers/state/CsrfStateItemHandler.scala

```diff
@@ -35,7 +35,7 @@ class CsrfStateItemHandler @Inject() (
    */
   override def item(implicit ec: ExecutionContext): Future[Item] = {
     idGenerator.generate.map { id =>
-      CsrfState(id)
+      CsrfState(cookieSigner.sign(id))
```

Comment:
This is not needed here, otherwise we sign the ID and then the complete state. The signing should be moved to the cookie creation,